src/PowerShellEditorServices/Session/PowerShellContext.cs

```diff
@@ -823,20 +823,11 @@ public Task ExecuteCommand(PSCommand psCommand)
 
         internal static TResult ExecuteScriptAndGetItem<TResult>(string scriptToExecute, Runspace runspace, TResult defaultValue = default(TResult))
         {
-            Pipeline pipeline = null;
-
-            try
+            using (var pwsh = PowerShell.Create())
             {
-                if (runspace.RunspaceAvailability == RunspaceAvailability.AvailableForNestedCommand)
-                {
-                    pipeline = runspace.CreateNestedPipeline(scriptToExecute, false);
-                }
-                else
-                {
-                    pipeline = runspace.CreatePipeline(scriptToExecute, false);
-                }
-
-                Collection<PSObject> results = pipeline.Invoke();
+                pwsh.Runspace = runspace;
+                Collection<PSObject> results = pwsh.AddScript(scriptToExecute)
+                    .Invoke(null, new PSInvocationSettings() { AddToHistory = false });
```

Comment:
`AddToHistory` is `false` by default, so you should be able to just use `Invoke()` here.

Also, there's a generic version of `Invoke` that specifies return type, so you can use `Invoke<TResult>()` and ditch the manual conversion code below.